.circleci/config.yml

```diff
@@ -1908,6 +1908,91 @@ jobs:
           paths:
             - dnsdist
 
+  build-dnsdist-with-tsan:
+    docker:
+      - image: debian:buster
+        auth:
+          username: powerdnsreadonly
+          password: $DOCKERHUB_PASSWORD
+        environment:
+          TSAN_OPTIONS: 'halt_on_error=1:suppressions=/opt/project/pdns/dnsdistdist/dnsdist-tsan.supp'
+    steps:
+      - run:
+          name: Install dependencies
+          command: |
+            apt-get update && apt-get -qq --no-install-recommends install \
+            autoconf \
+            automake \
+            g++ \
+            git \
+            libboost-all-dev \
+            libcap-dev \
+            libcdb-dev \
+            libedit-dev \
+            libfstrm-dev \
+            libh2o-evloop-dev \
+            liblmdb-dev \
+            libluajit-5.1-dev \
+            libre2-dev \
+            libsnmp-dev \
+            libsodium-dev \
+            libssl-dev \
+            libsystemd-dev \
+            libtool \
+            make \
+            pkg-config \
+            ragel \
+            systemd \
+            python3-venv
+      - install-clang-8
+      - restore-cache-ccache:
+          product: dnsdist-with-tsan
+      - get-workspace
+      - run:
+          name: autoconf
+          command: BUILDER_VERSION=0.0.0-git1 autoreconf -vfi
+          working_directory: ~/project/pdns/dnsdistdist
+      - run:
+          name: configure
+          command: |
+            CC='clang-8' \
+            CXX='clang++-8' \
+            CFLAGS="-O1 -Werror=vla -Werror=shadow -Wformat=2 -Werror=format-security -Werror=string-plus-int" \
+            CXXFLAGS="-O1 -Werror=vla -Werror=shadow -Wformat=2 -Werror=format-security -Werror=string-plus-int -Wp,-D_GLIBCXX_ASSERTIONS" \
+            ./configure \
+            --enable-option-checking=fatal \
+            --enable-unit-tests \
+            --enable-dnstap \
+            --enable-dnscrypt \
+            --enable-dns-over-tls \
+            --enable-dns-over-https \
+            --enable-systemd \
+            --prefix=/opt/dnsdist-with-tsan \
+            --with-libsodium \
+            --with-lua=luajit \
+            --with-libcap \
+            --with-re2 \
+            --enable-tsan
+          working_directory: ~/project/pdns/dnsdistdist
+      - run:
+          name: build
+          command: make -j1 -k
+          working_directory: ~/project/pdns/dnsdistdist
+      - save-ccache-cache:
+          product: dnsdist-with-tsan
+      - run:
+          name: Run unit tests
+          command: make check || (cat test-suite.log; false)
+          working_directory: ~/project/pdns/dnsdistdist
+      - run:
+          name: Install resulting binaries
+          command: make install
+          working_directory: ~/project/pdns/dnsdistdist
+      - persist_to_workspace:
+          root: /opt
+          paths:
+            - dnsdist
```

Comment:
I believe this would overwrite the non-tsan build (or vice versa, i.e., this TSAN PR introduces a race condition ;) )

Reply:
Ah, I was wondering about that! It's not very clear to me how `persist_to_workspace` works :-/

Reply:
The workspace is per workflow and both the tsan and non-tsan builds are in the `build-and-test-all` workflow. You could make a separate workflow (but I don't really see benefits to that), or take my previous suggestions.

Reply:
Ah, you already did that!

Reply:
I think I'm starting to see it, thanks! It looks like I also need to change the `persist_to_workplace` path to `dnsdist-with-tsan`, right?

Reply:
yes!